bundles/pubsub/pubsub_utils/src/pubsub_serializer_handler.c

```diff
@@ -0,0 +1,360 @@
+        free(existingVersion);
+        valid = false;
+    }
+
+    if (valid) {
+        celix_array_list_t *entries = hashMap_get(handler->serializationServices, (void *) (uintptr_t) msgId);
+        if (entries == NULL) {
+            entries = celix_arrayList_create();
+        }
+        pubsub_serialization_service_entry_t *entry = calloc(1, sizeof(*entry));
+        entry->svcId = svcId;
+        entry->properties = svcProperties;
+        entry->msgFqn = celix_utils_strdup(msgFqn);
+        entry->msgId = msgId;
+        entry->msgVersion = msgVersion;
+        entry->svc = svc;
+        celix_arrayList_add(entries, entry);
+        celix_arrayList_sort(entries, compareEntries);
+
+        hashMap_put(handler->serializationServices, (void *) (uintptr_t) msgId, entries);
+    } else {
+        celix_version_destroy(msgVersion);
+    }
+    celixThreadRwlock_unlock(&handler->lock);
+}
+
+void pubsub_serializerHandler_removeSerializationService(pubsub_serializer_handler_t* handler, pubsub_message_serialization_service_t* svc, const celix_properties_t* svcProperties) {
+    long svcId = celix_properties_getAsLong(svcProperties, OSGI_FRAMEWORK_SERVICE_ID, -1L);
+    const char *msgFqn = celix_properties_get(svcProperties, PUBSUB_MESSAGE_SERIALIZATION_SERVICE_MSG_FQN_PROPERTY, NULL);
+    uint32_t msgId = (uint32_t)celix_properties_getAsLong(svcProperties, PUBSUB_MESSAGE_SERIALIZATION_SERVICE_MSG_ID_PROPERTY, 0L);
+    if (msgId == 0) {
+        msgId = celix_utils_stringHash(msgFqn);
+    }
+
+    celixThreadRwlock_writeLock(&handler->lock);
+    celix_array_list_t* entries = hashMap_get(handler->serializationServices, (void*)(uintptr_t)msgId);
+    if (entries != NULL) {
+        pubsub_serialization_service_entry_t *found = NULL;
+        for (int i = 0; i < celix_arrayList_size(entries); ++i) {
+            pubsub_serialization_service_entry_t *entry = celix_arrayList_get(entries, i);
+            if (entry->svcId == svcId) {
+                found = entry;
+                celix_arrayList_removeAt(entries, i);
+                celix_arrayList_sort(entries, compareEntries);
+                break;
+            }
+        }
+        if (found != NULL) {
+            free(found->msgFqn);
+            celix_version_destroy(found->msgVersion);
+            free(found);
+        }
+        if (celix_arrayList_size(entries) == 0) {
+            hashMap_remove(handler->serializationServices, (void*)(uintptr_t)msgId);
+            celix_arrayList_destroy(entries);
+        }
+    }
+    celixThreadRwlock_unlock(&handler->lock);
+}
+
+celix_status_t pubsub_serializerHandler_serialize(pubsub_serializer_handler_t* handler, uint32_t msgId, const void* input, struct iovec** output, size_t* outputIovLen) {
```

Comment:
Because struct iovec** output is an array iovecs. I added the message serialization service doc as used in the pubsub serialization handler doc, maybe this makes it more clearer.